smartsim/settings/pbsSettings.py

```diff
@@ -185,40 +213,71 @@ def format_batch_args(self) -> t.List[str]:
+        if has_nodes and not isinstance(has_nodes, int):
+            raise TypeError("The value for 'nodes' must be an integer")
+
+        for key, value in checked_resources.items():
+            allowed_types = [int, str]
+            if not any(isinstance(key, type) for type in allowed_types):
+                raise TypeError(
+                    f"The type of {key=} is {type(key)}. Only int and str "
+                    "are allowed."
+                )
+            if not any(isinstance(value, type) for type in allowed_types):
+                raise TypeError(
+                    f"The value associated with {key=} is {type(value)}. Only int "
+                    "and str are allowed."
+                )
+
     def _create_resource_list(self) -> t.List[str]:
+
+        self._sanity_check_resources()
         res = []
 
-        # get select statement from resources or kwargs
-        if "select" in self.resources:
-            res += [f"-l select={str(self.resources['select'])}"]
-        else:
-            select = "-l select="
-            if self._nodes:
-                select += str(self._nodes)
-            else:
-                raise SmartSimError(
-                    "Insufficient resource specification: no nodes or select statement"
-                )
-            if self._ncpus:
-                select += f":ncpus={self._ncpus}"
-            if self._hosts:
-                hosts = ["=".join(("host", str(host))) for host in self._hosts]
-                select += f":{'+'.join(hosts)}"
-            res += [select]
-
-        if "place" in self.resources:
-            res += [f"-l place={str(self.resources['place'])}"]
-        else:
-            res += ["-l place=scatter"]
+        # Pop off some specific keywords that need to be treated separately
+        resources = self.resources # Note this is a copy so not modifying original
 
-        # get time from resources or kwargs
-        if "walltime" in self.resources:
-            res += [f"-l walltime={str(self.resources['walltime'])}"]
+        # Construct the basic select/nodes statement
+        if select := resources.pop("select", None):
+            select_command = f"-l select={select}"
+        elif nodes := resources.pop("nodes", None):
+            select_command = f"-l nodes={nodes}"
         else:
-            if self._time:
-                res += [f"-l walltime={self._time}"]
+            raise SSConfigError(
+                "Insufficient resource specification: no nodes or select statement"
+            )
+        if self._ncpus:
+            select_command += f":ncpus={self._ncpus}"
+        if self._hosts:
+            hosts = ["=".join(("host", str(host))) for host in self._hosts]
+            select_command += f":{'+'.join(hosts)}"
+        res += [select_command]
+
+        # All other "standard" resource specs
+        for resource, value in resources.items():
+            res += [f"-l {resource}={str(value)}"]
```

Comment:
nit: no need to make a `str` the f-str will do that implicitly:
```suggestion
            res += [f"-l {resource}={value}"]
```